listener/listener.go

```diff
@@ -237,12 +237,12 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 	// Create a handle class that has that function
 	listenerHandle := subReceiver.Listen(ctx, servicebus.HandlerFunc(
 		func(ctx context.Context, msg *servicebus.Message) error {
-			originalHandler := handler
-			for !message.IsDone(handler) {
-				handler = handler.Do(ctx, originalHandler, msg)
+			currentHandler := handler
```

Comment:
is there a unittest we can augment to catch this if we regress?

Reply:
Or write integration tests. They are easy to run too. The readme has a step by step to setup the environment using 2 make targets.

Reply:
Added integration tests to publish and listen a message twice.

It failed without the fix, and passed when we have the fix.